packages/legend-query-builder/src/components/__tests__/QueryBuilderVariables.test.tsx

```diff
@@ -0,0 +1,228 @@
+/**
+ * Copyright (c) 2020-present, Goldman Sachs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { test, expect } from '@jest/globals';
+import {
+  waitFor,
+  fireEvent,
+  getByTitle,
+  act,
+  getByText,
+} from '@testing-library/react';
+import { TEST_DATA__simpleProjection } from '../../stores/__tests__/TEST_DATA__QueryBuilder_Generic.js';
+import { TEST_DATA__ModelCoverageAnalysisResult_ComplexRelational } from '../../stores/__tests__/TEST_DATA__ModelCoverageAnalysisResult.js';
+import TEST_DATA__ComplexRelationalModel from '../../stores/__tests__/TEST_DATA__QueryBuilder_Model_ComplexRelational.json';
+import { integrationTest } from '@finos/legend-shared';
+import {
+  create_RawLambda,
+  PrimitiveType,
+  stub_RawLambda,
+} from '@finos/legend-graph';
+import { QUERY_BUILDER_TEST_ID } from '../QueryBuilder_TestID.js';
+import { TEST__setUpQueryBuilder } from '../QueryBuilderComponentTestUtils.js';
+import { CUSTOM_DATE_PICKER_OPTION } from '../shared/CustomDatePicker.js';
+import { variableExpression_setName } from '../../stores/shared/ValueSpecificationModifierHelper.js';
+
+test(
+  integrationTest(
+    'Query builder shows validation error for parameter name if existing duplicate constant name',
+  ),
+  async () => {
+    const { renderResult, queryBuilderState } = await TEST__setUpQueryBuilder(
+      TEST_DATA__ComplexRelationalModel,
+      stub_RawLambda(),
+      'model::relational::tests::simpleRelationalMapping',
+      'model::MyRuntime',
+      TEST_DATA__ModelCoverageAnalysisResult_ComplexRelational,
+    );
+    await act(async () => {
+      queryBuilderState.initializeWithQuery(
+        create_RawLambda(
+          TEST_DATA__simpleProjection.parameters,
+          TEST_DATA__simpleProjection.body,
+        ),
+      );
+      queryBuilderState.setShowParametersPanel(true);
```

Comment:
Can we show parameter/constant panel mirroring how the user would do it. 
I.e go to advanced options -> click show parameters/constants. 
We can even move those out to helper query builder test functions